DotNetHowToAutocomplete/DotNetHowToAutocomplete/Controllers/HomeController.cs

```diff
@@ -0,0 +1,159 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Web;
+using System.Web.Mvc;
+
+using Microsoft.Azure.Search;
+using Microsoft.Azure.Search.Models;
+using System.Configuration;
+
+namespace AutocompleteTutorial.Controllers
+{
+    public class HomeController : Controller
+    {
+        private static SearchServiceClient _searchClient;
+        private static ISearchIndexClient _indexClient;
+        private static string IndexName = "nycjobs";
+
+        public static string errorMessage;
+
+        private void InitSearch()
+        {
+                string searchServiceName = ConfigurationManager.AppSettings["SearchServiceName"];
+                string apiKey = ConfigurationManager.AppSettings["SearchServiceApiKey"];
+
+                // Create a reference to the NYCJobs index
+                _searchClient = new SearchServiceClient(searchServiceName, new SearchCredentials(apiKey));
+                _indexClient = _searchClient.Indexes.GetClient(IndexName);
+        }
+
+        public ActionResult Index()
+        {
+            return View();
+        }
+
+        public ActionResult IndexJavaScript()
+        {
+            return View();
+        }
+
+        public ActionResult About()
+        {
+            ViewBag.Message = "Your application description page.";
+
+            return View();
+        }
+
+        public ActionResult Contact()
+        {
+            ViewBag.Message = "Your contact page.";
+
+            return View();
+        }
+        public ActionResult Suggest(bool highlights, bool fuzzy, string term)
+        {
+            InitSearch();
+
+            // Call suggest API and return results
+            SuggestParameters sp = new SuggestParameters()
+            {
+                UseFuzzyMatching = fuzzy,
+                Top = 5
+            };
+
+            if (highlights)
+            {
+                sp.HighlightPreTag = "<b>";
+                sp.HighlightPostTag = "</b>";
+            }
+
+            DocumentSuggestResult resp = _indexClient.Documents.Suggest(term, "sg", sp);
```

Comment:
Not sure how much we care about showing best practices here... If you think that's important, I'd recommend calling `SuggestAsync` instead and make the controller method async too.